sycl/include/sycl/ext/oneapi/bindless_images.hpp

```diff
@@ -910,6 +911,29 @@ DataT fetch_image(const sampled_image_handle &imageHandle [[maybe_unused]],
 #endif
 }
 
+template <typename DataT>
+#ifdef __SYCL_DEVICE_ONLY__
+[[__sycl_detail__::__uses_aspects__(
+    sycl::aspect::ext_oneapi_bindless_images_gather)]]
+#endif
+std::enable_if_t<std::is_same_v<DataT, float4> || std::is_same_v<DataT, int4> ||
+                     std::is_same_v<DataT, uint4>,
+                 DataT> gather_image(const sampled_image_handle &imageHandle
+                                     [[maybe_unused]],
+                                     const float2 &coords [[maybe_unused]],
+                                     const unsigned i [[maybe_unused]]) {
+#if defined(__SYCL_DEVICE_ONLY__)
+#if defined(__NVPTX__)
+  return __invoke__SampledImageGather<DataT>(
+      CONVERT_HANDLE_TO_SAMPLED_IMAGE(imageHandle.raw_handle, float2::size()),
+      coords, i);
+#endif
```

Comment:
We still need a return value of the other paths, even if they won't ever be run:
```suggestion
#else
  return {0,0,0,0};
#endif
```